snowpack/src/config.ts

```diff
@@ -40,6 +40,7 @@ const ALWAYS_EXCLUDE = ['**/node_modules/**/*', '**/web_modules/**/*', '**/.type
 
 // default settings
 const DEFAULT_CONFIG: SnowpackUserConfig = {
+  root: process.cwd(),
```

Comment:
This looks good! This looks like the correct default behavior to me.

I think in addition to this line, we also need to add resolution below in the `loadConfigurationForCLI()` method when we’re loading a cosmiconfig. For example, say a user has their project set up like so:

```
├── config/
│   └── snowpack.config.js
└── src/
```

And say in their config they set:

```
{
  root: '../src',
}
```

**We should resolve this from the location of `config/snowpack.config.js`.** As is, it’s unclear how `src` will resolve. And since we do want to eventually follow this behavior, treating your config file location as root (similar to TSConfig, etc.), I think it’s better to do it in the same PR where we introduce the option. That way we won’t break setups for people if they start using this now, and we change that behavior later.

Reply:
I already started working on `loadConfigurationForCLI()` but I got a little stuck because I wasn't sure whether to add the `root` as a flag to the CLI too. I'll work on it the next days and update the MR to get your opinion.

Reply:
@drwpow I'd actually like to tackle that problem in a separate PR, if you don't mind. It's a bigger ask than I think you realize. We have lots of other properties in the config that also need this support that are currently always relative to `process.cwd()` (`mount`, `plugins`, `alias`, all file paths like `out`, etc.). 

We also need to decide if we continue to support `extends`. If we do, we'll also need to handle this support relative to the extended config. But, I think since we've begun leaning more on JS config than JSON lately, I think we could also just skip this work and remove `extends` support in v3 in favor just whatever you want to script in JS.

Last reason: it means that most existing tests will continue to work as-is in this PR. Given that this touches so many different parts of the codebase, test stability is a plus at the moment.

We can still keep thrash down by only merging this into the v3 release branch.